programs/stake/src/stake_state.rs

```diff
@@ -757,33 +761,71 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
             match self.state()? {
                 StakeState::Stake(meta, mut stake) => {
                     meta.authorized.check(signers, StakeAuthorize::Staker)?;
+                    let split_rent_exempt_reserve = calculate_split_rent_exempt_reserve(
+                        meta.rent_exempt_reserve,
+                        self.data_len()? as u64,
+                        split.data_len()? as u64,
+                    );
 
                     // verify enough lamports for rent in new stake with the split
                     if split.lamports()? + lamports < meta.rent_exempt_reserve
-                     // verify enough lamports left in previous stake and not full withdrawal
+                    // verify enough lamports left in previous stake and not full withdrawal
                         || (lamports + meta.rent_exempt_reserve > self.lamports()? && lamports != self.lamports()?)
+                    // verify that the same rent_exempt_reserve is sufficient to make the new
+                    // stake account rent exempt needed because the new uninitialized may be any
+                    // size larger than std::mem::size_of::<StakeState>()
+                        || split_rent_exempt_reserve > meta.rent_exempt_reserve
```

Comment:
this new error condition will make it fail to split smaller-sized stake account into bigger-sized stake account even if there is no inconsistency between `rent_exempt_reserve` and `data_len()`. That will be very hard to debug for users. If could, I'd like to avoid this by adjusting split_stake and split_rent_exempt_reserve...

Reply:
Yes, my intent was to prevent all splits from smaller-sized stake accounts into bigger-sized ones, since we are otherwise addressing the causes of small stake accounts with incorrect larger `rent_exempt_reserve`s.

> If could, I'd like to avoid this by adjusting split_stake and split_rent_exempt_reserve...

I'm sorry, I don't understand what you mean. Unless you are referring to your comment above?